### PR TITLE
feat: Astro support

### PR DESCRIPTION
**Pull request by feugy**
### 📓 What's in there?

This is an initial draft for Astro first-class support.

`beforeSend` is supported by using  `speedInsightsBeforeSend` as global window function
```
<script is:inline>
  function speedInsightsBeforeSend(data){
    console.log("Capured event!")
    return data;
  }
</script>
```

TODO list:
- [ ] use astro@4
- [ ] documentation
- [ ] check types and autocompletion

### 🧪 How to test?

There's a new example app, based on the Astro official blog template
- get the dependencies `pnpm i`
- update `apps/astro/src/components/BaseHead.astro` to set `scriptSrc="https://va.vercel-scripts.com/v1/speed-insights/script.js"` on the speed insights component. This will attempt to report metrics so we can see the reported payloads.
- `cd apps/astro && pnpm dev` to start the app
- visit the app, and navigate its pages.

### CODE REVIEW

**Comment by MoustaphaDev** [comment]
Agreed that using a component is better DX!

I'm exploring an alternative approach to address the `beforeSend` as props limitation that might be more suitable than excluding that future (in my opinion), although not ideal.

Here's how it would work and how it would be documented:
> **Note:** Because of some limitations (blah blah), if you want to use the `beforeSend` feature, you can define a `speedInsightsBeforeSend` function in an inline script in your Astro page or layout:
> ```astro
> <script is:inline>
>   function speedInsightsBeforeSend(data){
>     console.log("Capured event!")
>     return data;
>   }
> </script>
> ``` 

Here's how the function will be used internally:
```ts
// ...
const beforeSend = window.speedInsightsBeforeSend;
injectSpeedInsights({ route, ...props, framework: 'astro', beforeSend });
// ...
```

Let me know what you think 🙂

**Comment by tobiaslins** [comment]
@MoustaphaDev thanks for the review!
Thats a great idea. I just updated the code to look for `window.speedInsightsBeforeSend` to support that!

File: apps/astro/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "build": "astro build",
+    "dev": "astro dev",
+    "preview": "astro preview",
+    "start": "astro dev"
+  },
+  "dependencies": {
+    "@astrojs/mdx": "^1.1.5",
+    "@astrojs/rss": "^3.0.0",
+    "@astrojs/sitemap": "^3.0.3",
+    "astro": "^3.6.3"
```
Comment (MoustaphaDev) [inline apps/astro/package.json:16]: Nit: This is just example project, but Astro 4 came out a week ago, it wouldn't hurt [upgrading the dependencies](https://github.com/withastro/astro/blob/main/examples/blog/package.json#L14-L17)
```suggestion
    "@astrojs/mdx": "^2.0.1",
    "@astrojs/rss": "^4.0.1",
    "@astrojs/sitemap": "^3.0.3",
    "astro": "^4.0.4"
```

File: apps/astro/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  "devDependencies": {
+    "@vercel/speed-insights": "workspace:*"
```
Comment (MoustaphaDev) [inline apps/astro/package.json:19]: This should be a dependency, right?

**Review by MoustaphaDev — APPROVED** [review]
Code looks good to me, I just have two small suggestions

**Review by FredKSchott — COMMENTED** [review]
LGTM!

**Comment by tobiaslins** [comment]
Updated to latest versions :) Thanks all <picture data-single-emoji=":lfg-intense:" title=":lfg-intense:">:lfg-intense:</picture>

**Review by chriswdmr — APPROVED** [review]

**Pull request merged**